src/bridge/cockpitpipechannel.c

```diff
@@ -384,6 +407,10 @@ cockpit_pipe_channel_prepare (CockpitChannel *channel)
       else
         self->pipe = cockpit_pipe_spawn ((const gchar **)argv, (const gchar **)env, dir, flags);
     }
+  else if (internal && lookup_internal_fd (internal, &fd))
+    {
+      self->pipe = cockpit_pipe_new (internal, fd, fd);
```

Comment:
This specifies the same fd for both reading and writing, which is error prone and misleading. Unix pipes are unidirectional, and the sender and receiver need to agree which one gets passed and which one the sender keeps. This is related to my question above how we keep track of reading and writing fds.